accounts/abi/bind/backends/simulated.go

```diff
@@ -529,9 +553,30 @@ func (b *SimulatedBackend) EstimateGas(ctx context.Context, call ethereum.CallMs
 // state is modified during execution, make sure to copy it if necessary.
 func (b *SimulatedBackend) callContract(ctx context.Context, call ethereum.CallMsg, block *types.Block, stateDB *state.StateDB) (*core.ExecutionResult, error) {
 	// Ensure message is initialized properly.
-	if call.GasPrice == nil {
+	// EIP1559 guards
+	// If we have finalized EIP1559 and do not have a properly formed EIP1559 trx, sub in default values
+	eip1559 := b.config.IsEIP1559(block.Number())
+	eip1559Finalized := b.config.IsEIP1559Finalized(block.Number())
+	if eip1559Finalized && (call.MaxMinerBribePerGas == nil || call.FeeCapPerGas == nil || call.GasPrice != nil) {
```

Comment:
Maybe it is better to judge eip1559 and eip1559Finalized in one `if` clause before modifying the attributes value of call, e.g. `if !eip1559 && eip1559Finalized`, if it is true, throw an error directly.